translator/internaldata/oc_to_traces.go

```diff
@@ -38,8 +38,8 @@ func OCToTraceData(td consumerdata.TraceData) data.TraceData {
 
 	if len(td.Spans) == 0 {
 		// At least one of the td.Node or td.Resource is not nil. Set the resource and return.
-		traceData.SetResourceSpans(data.NewResourceSpansSlice(1))
-		ocNodeResourceToInternal(td.Node, td.Resource, traceData.ResourceSpans().Get(0))
+		traceData.ResourceSpans().Resize(1)
```

Comment:
I think it would be better to use a temp var for `traceData.ResourceSpans()` to make sure it is evaluated only once.

Reply:
done.